test/runnable/alias_tupleof.d

```diff
@@ -0,0 +1,21 @@
+struct F
+{
+    alias thisTup = typeof(this).tupleof;
```

Comment:
I suggest changing this example to:
```suggestion
    alias thisTup = this.tupleof;
```

as accessing `this.{a, b, c, d}` through a type looks quite confusing. I don't know if it's late to change this, but I'd expect that only `this.tupleof` (or `.tupleof` on any object instance (class or struct, doesn't matter)) would be able  to access instance members.

`typeof(this).tupleof` should, of course, continue working, but I expect it to only yield a sequnece of types, (and their names and default values) and not a direct reference to struct/class object members.

Reply:
No I did the right thing, what you propose is even an error since a few release (this or super used as a type)

>  Error: Using `this` as a type is obsolete. Use `typeof(this)` instead

Reply:
BTW when this is required, the compiler can rewrite a `VarExp` /`CallExp` to a `DotVarExp` that adds the  missing `this.`

Reply:
> Error: Using this as a type is obsolete. Use typeof(this) instead

I'm aware of https://github.com/dlang/dmd/pull/9711/files, however I think it's a bug to get this error here, as in this context, `typeof(this).tupleof` should be equivalent to `F.tupleof`, which I don't see how it could be allowed to reference instance (non-static) members. Whether you access `typeof(this)` inside of a class or struct, or `F.tupleof` (where `F` is their name for example) from outside must have the same semantics in the same module (I don't want 
to get into private / protected, but for now even in different modules).

Reply:
> I don't see how it could be allowed to reference instance (non-static) members

but .tupleof doesn't even return static members